event_model/documents/run_start.py

```diff
@@ -0,0 +1,203 @@
+from typing import Any, Dict, List, Union
+
+from typing_extensions import Annotated, Literal, NotRequired, TypedDict
+
+from event_model.documents.generate.type_wrapper import Field, add_extra_schema
+
+from .generate.type_wrapper import AsRef
+
+
+class Hints(TypedDict):
+    """Start-level hints"""
+
+    dimensions: NotRequired[
+        Annotated[
+            List[List[Union[List[str], str]]],
+            Field(
+                description="The independent axes of the experiment. "
+                "Ordered slow to fast",
+            ),
+        ]
+    ]
+
+
+class Calculation(TypedDict):
+    args: NotRequired[List]
+    callable: Annotated[
+        str, Field(description="callable function to perform calculation")
+    ]
+    kwargs: NotRequired[
+        Annotated[Dict[str, Any], Field(description="kwargs for calcalation callable")]
+    ]
+
+
+class Projection(TypedDict):
+    """Where to get the data from"""
+
+    calculation: NotRequired[
+        Annotated[
+            Calculation,
+            Field(
+                description="required fields if type is calculated",
+                title="calculation properties",
+            ),
+        ]
+    ]
+    config_index: NotRequired[int]
+    config_device: NotRequired[str]
+    field: NotRequired[str]
+    location: NotRequired[
+        Annotated[
+            Literal["start", "event", "configuration"],
+            Field(
+                description="start comes from metadata fields in the start document, "
+                "event comes from event, configuration comes from configuration "
+                "fields in the event_descriptor document"
+            ),
+        ]
+    ]
+    stream: NotRequired[str]
+    type: NotRequired[
+        Annotated[
+            Literal["linked", "calculated", "static"],
+            Field(
+                description="linked: a value linked from the data set, "
+                "calculated: a value that requires calculation, "
+                "static:  a value defined here in the projection ",
+            ),
+        ]
+    ]
+    value: NotRequired[
+        Annotated[
+            Any,
+            Field(
+                description="value explicitely defined in the projection "
+                "when type==static."
+            ),
+        ]
+    ]
+
+
+RUN_START_EXTRA_SCHEMA = {
+    "definitions": {
+        "Projection": {
+            "allOf": [
+                {
+                    "if": {
+                        "allOf": [
+                            {"properties": {"location": {"enum": "configuration"}}},
```

Comment:
These seems to be `... "enum": ["configuration"] ...` in the original doc, not sure if this is equivalent?

Reply:
Must have been lost when it was brought over, I'll add it!